test/sneaker_handlers/retry_handler_test.rb

```diff
@@ -0,0 +1,51 @@
+require "test_helper"
+
+class SneakersHandlers::RetryHandlerTest < Minitest::Test
+  extend MiniTest::Spec::DSL
+  let(:channel) { Minitest::Mock.new() }
+  let(:queue) {
+    q = Minitest::Mock.new
```

Comment:
Not a blocker, but you can about the naked `q` return at the end (and lifewise the local variable assignment) with a tap

```ruby
let(:queue)  { 
  Minitest::Mock.new.tap { |q| q.expect :name, "test.queue" } 
} 
```